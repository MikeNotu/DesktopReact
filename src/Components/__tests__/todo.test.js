import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../todo";

afterEach(() => {
    cleanup();
});

test("should render non-completed todo component", () => {
    const todo = { id: 1, title: "wash dishes", completed: false };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash dishes");
    expect(todoElement).not.toContainHTML("<strike>");
});

test("should render completed todo component", () => {
    const todo = { id: 2, title: "wash car", completed: true };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-2");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash car");
    expect(todoElement).toMatchSnapshot("<strike>");
});
