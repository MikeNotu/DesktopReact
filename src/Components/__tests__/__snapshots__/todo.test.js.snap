// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render completed todo component: <strike> 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
     
    <h1>
      wash car
    </h1>
     
  </strike>
</div>
`;
